src/binder/Binder.ts

```diff
@@ -0,0 +1,70 @@
+import { Annotations, BindingAnnotation } from '../annotations';
+import Reflector from '../reflect';
+import { Binding } from './Binding';
+import { LinkedBinder } from './LinkedBinder';
+import { MultiBinder } from './MultiBinder';
+
+export interface Module extends Object {
+  configure?(binder: Binder): void;
+}
+
+export class Binder {
+  _bindings = new Map<string, Binding>();
+  _modules: Array<Module> = [];
+  _multiBinders?: Record<string, MultiBinder>;
+  _parent?: Binder;
+
+  constructor(modules: Array<Module> = [], parent?: Binder) {
+    this._parent = parent;
+    modules.forEach((m) => this.install(m));
+  }
+
+  bindKey(key: string): LinkedBinder {
+    const binding = new Binding(key);
+    this._bindings.set(key, binding);
+
+    return new LinkedBinder(binding);
+  }
+
+  bind(_class: any) {
+    throw new Error('"bind" is not supported');
+  }
+
+  setParent(parent: Binder) {
+    this._parent = parent;
+  }
+
+  getBinding(key: string): Binding | undefined {
+    let binding = this._bindings.get(key);
+
+    if (!binding && this._parent) {
+      binding = this._parent.getBinding(key);
+    }
+
+    return binding;
+  }
+
+  install(module: Module): void {
+    if (this._modules.indexOf(module) !== -1) {
```

Comment:
```suggestion
    if (this._modules.find(module)) {
```

Reply:
That would only work if we provided a callback instead.

```ts
    if (this._modules.find((m) => m === module)) {
```

I think `indexOf` is fine.